### ISSUE TITLE
Brackets freezes when installing/uninstalling extensions (and other situations) and doesn't respond on Windows 7 64-bit

### ISSUE THREAD

**Issue by AlexanderJ2357** [adobe/brackets#11477]
Brackets freezes a lot on windows 7 (I have it on OSX too and haven't noticed freezes there but I don't use OSX as much as windows). It freezes especially when I am installing/uninstalling extensions.

Another situation in which it freezes is when I open the Extension Manager and try to close it by clicking on the Close button while it is loading.

I am afraid that I lose unsaved work when it freezes. I hope it is possible to fix this issue.

**Comment by hussainb** [comment]
+1.
This issue was intermittent before the  build 1.4.0-16380, but after upgrading to build 1.4.0-16380 it happens always, the same happens when pressing Reloading(F5) while developing extensions.
Have to use task manager to kill brackets every time and this is very irritating.

**Comment by nethip** [comment]
@AlexanderJ2357 @hussainb Do you have any extensions installed? Just to rule out if any extension is creating the issue, could you rename the preferences folder, and relaunch brackets?

**Comment by AlexanderJ2357** [comment]
I do have a few of the popular extensions installed like beautify, emmet etc.
I read that a lot more people are experiencing crashing issues with Brackets, especially with large files.
I hope the issues are fixed in the future but for now, I am use Atom and happy with it.

**Comment by hussainb** [comment]
@nethip 
I renamed the user folder under the extensions folder and opened brackets, It created a new empty user folder and then I tried to reload(F5). This time too brackets crashed. I tried to trace errors in the console but when brackets crashes, the developer tools crashes too.
This now doesn't look like an extension issue to me.

The only thing I noticed that if no file is open in the editor (can see the 'CODE THE WEB' logo and project folder open in sidebar) the reload doesn't hang brackets.

System Info
Windows 7 64-bit

**Comment by nethip** [comment]
@hussainb Could you disable our instant search and see if this problem goes away. Use the preferences below to do that. 

```
"findInFiles.instantSearch": false,
"findInFiles.nodeSearch": false
```

Also before reloading, give it a few seconds before Brackets gets completely initialised.

**Comment by hussainb** [comment]
@nethip 
I decided to debug brackets by myself, What I found is that the function `browserReload(href)` in `DocumentCommandHandlers.js` needed a timeout before it updated the href at line no 1638.

The url :
https://github.com/adobe/brackets/blob/389d9e638c66fd1b65a1972a90acaaaa572bb8ab/src/document/DocumentCommandHandlers.js#L1638

The issue didn't re-appear once I added the timeout of 1000ms in the minified `main.js` in my local bracket installation.

**Comment by nethip** [comment]
@hussainb That's a good workaround as of now. But it would be great if can try the steps I had mentioned above, about disabling the node search, and tell us if it fixes the problem. This could definently help the community to have a workaround, in case they run into this issue. Thanks!

**Comment by hussainb** [comment]
@nethip I am sorry I forgot to mention that I tried disabling the instantSearch and nodeSearch the way you stated above, and it did not do any good.

Also, I think the issue is not related to the newly implemented instant search feature since I faced the problem intermittently (1 in 5 reloads) in the older version which did not have instant search.
I was able to live with that before, but after this upgrade the issue has become consistent.

**Comment by nethip** [comment]
@hussainb That is a huge relief! So as of now having a `timeout` seems to be the only solution. Thanks for the workaround.

**Comment by abose** [comment]
@hussainb as another workaround, if you have brackets dev console open, When it crashes,
switch to the dev console> click on the close button, and then press f5(reload) on the dev console, and brackets will reload properly.
![image](https://cloud.githubusercontent.com/assets/5336369/8950149/07249f74-35db-11e5-8210-242a314175b7.png)

**Comment by nethip** [comment]
@abose That is a good work around too! But I think the issue with brackets handing is popping up even without the devtools open.

**Comment by abose** [comment]
Yes, It is just a temporary workaround!

**Comment by hussainb** [comment]
@nethip  Yes, brackets hangs even when the dev tools is not open, and since I shifted all my dev work to brackets IDE, I needed to fix it at code level.

**Comment by nethip** [comment]
@hussainb Would you mind putting a PR for this?

**Comment by hussainb** [comment]
@nethip , Will take some time though, But will do it for sure.

**Comment by nethip** [comment]
No issues :+1: Take your time.

**Comment by MusicalCreeper01** [comment]
I'm getting the same issue when installing and uninstalling extensions. It works fine on my Mac book, but not my main desktop computer.

I'm running the latest version of Brackets (Release 1.4 build 1.4.0-16380), and Windows 8.1 pro.
